alf/environments/wrappers.py

```diff
@@ -12,187 +12,166 @@
-        def _traverse_append(result, data, path_prefix=""):
-            assert isinstance(result, OrderedDict)
-            if path_prefix:
-                path_prefix += "."
-            for key, field in data.items():
-                path = path_prefix + key
-                if isinstance(field, dict):
-                    if key not in result:
-                        result[key] = OrderedDict()
-                    result[key] = _traverse_append(result[key], field, path)
-                else:
-                    if key not in result:
-                        result[key] = deque(maxlen=self._stack_size)
-                    result[key].append(field)
-            return result
-
-        def _traverse_stack(result, path_prefix=""):
-            if not path_prefix or isinstance(result, OrderedDict):
-                for key, field in result.items():
-                    if path_prefix:
-                        path_prefix += "."
-                    path = path_prefix + key
-                    result[key] = _traverse_stack(field, path)
-            else:
-                if not self._fields_to_stack or (
-                        path_prefix in self._fields_to_stack):
-                    space = self._get_space_for_path(path_prefix)
-                    if isinstance(space, gym.spaces.Box):
-                        if (self._channel_order == 'channels_last'):
-                            result = np.concatenate(result, axis=-1)
-                        else:
-                            result = np.concatenate(result, axis=0)
-                    elif isinstance(space, gym.spaces.MultiDiscrete):
-                        result = np.concatenate(result, axis=-1)
-                    else:
-                        assert False, ("space with path {} not recognized: " +
-                                       "{}").format(path_prefix, str(space))
-                else:  # not stacking field
-                    result = result[-1]  # take last/current frame
-            return result
-
-        if isinstance(self.observation_space, gym.spaces.Dict):
-            result = OrderedDict()
-            for frame in self._frames:
-                result = _traverse_append(result, frame)
-            result = _traverse_stack(result)
-            return result
-        else:  # Assuming input is image space
-            # Always stacks regardless of _fields_to_stack
-            if self._channel_order == 'channels_last':
-                return np.concatenate(self._frames, axis=-1)
-            else:
-                return np.concatenate(self._frames, axis=0)
+            raise ValueError("Unsupported space:%s" % observation_space)
+
+    def transform_observation(self, observation, fields=None):
+        queue = self._frames.get(fields, None)
+        if not queue:
+            queue = deque(maxlen=self._stack_size)
+            for _ in range(self._stack_size):
+                queue.append(observation)
+            self._frames[fields] = queue
```

Comment:
can fields be None here?  If so, index to _frames will be None?

Reply:
`None` can be  a key value for `dict`,  and the field will never be None here (empty str when observation is not nested )